crates/store/re_log_encoding/src/decoder/streaming.rs

```diff
@@ -0,0 +1,371 @@
+use std::pin::Pin;
+
+use bytes::{Buf, BytesMut};
+use re_build_info::CrateVersion;
+use re_log::external::log::warn;
+use re_log_types::LogMsg;
+use tokio::io::{AsyncBufRead, AsyncReadExt};
+use tokio_stream::Stream;
+
+use crate::{
+    codec::file::{self},
+    Compression, EncodingOptions, VersionPolicy,
+};
+
+use super::{read_options, DecodeError, FileHeader};
+
+pub struct StreamingDecoder<R: AsyncBufRead> {
+    version: CrateVersion,
+    options: EncodingOptions,
+    reader: R,
+    // buffer used for uncompressing data. This is a tiny optimization
+    // to (potentially) avoid allocation for each (compressed) message
+    uncompressed: Vec<u8>,
+    // internal buffer for unprocessed bytes
+    unprocessed_bytes: BytesMut,
+    // flag to indicate if we're expecting more data to be read.
```

Comment:
No reason not to make these comments actual docstrings so they actually show up in all the tools.